lib/resources/odata.js

```diff
@@ -39,7 +39,7 @@ module.exports = (service, endpoint) => {
 
     // serves filtered single-row data.
     //const singleRowRegex = /^\/projects\/(\d+)\/forms\/([a-z0-9-_]+).svc\/Submissions\((?:'|%27)((?:uuid:)?[a-z0-9-]+)(?:'|%27)\)(\/.*)*$/i;
-    const uuidRegex = /^(?:'|%27)((?:uuid:)?[a-z0-9-]+)(?:'|%27)$/i;
+    const uuidRegex = /^'((?:uuid:)?[a-z0-9-]+)'$/i;
```

Comment:
When I first started working on this PR, I thought I had to add `%3A` to this regular expression for `:`, but the change needed turned out to be elsewhere. After taking a look at this endpoint, I'm pretty sure that we don't need `%27` in the regular expression. `uuidRegex` is only used in `getUuid()`, which is passed `params.uuid`, but Express already ensures that `params.uuid` is decoded. My guess is that this regular expression used to be part of the route itself, not used for the param, and the single quote needed to be encoded there.

Reply:
yes, that is the exact history there.